### PR TITLE
Remove support for php 5.4 & php 5.5

### PR DESCRIPTION
**Pull request by NiR-**
This change is [Reviewable](https://reviewable.io/reviews/knplabs/gaufrette/503)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
         }
     ],
     "require": {
-        "php": ">=5.4"
+        "php": ">=5.6",
+        "akeneo/phpspec-skip-example-extension": "~1.2"
```
Comment (Shivoham) [inline composer.json]: It should be a dev requirement

File: tests/Gaufrette/Functional/Adapter/AzureMultiContainerBlobStorageTest.php [hunk 2]
```diff
@@ -182,14 +200,28 @@ public function shouldKeepFileObjectInRegister()
...
         $FileObjectA->setContent('ABC');
 
-        $FileObjectB = $this->filesystem->createFile('container11/somefile');
+        $FileObjectB = $this->filesystem->createFile($path);
         $FileObjectB->setContent('DEF');
 
         $this->assertEquals('DEF', $FileObjectB->getContent());
+    }
+
+    private function getContainerName($prefix)
```
Comment (Shivoham) [inline tests/Gaufrette/Functional/Adapter/AzureMultiContainerBlobStorageTest.php]: It's not obivous this method won't return the same result when called with the same args. I'd prefer something like `createUniqueContainerName($prefix)`

**Comment by NiR-** [comment]
@dkarlovi From time to time `AzureMultiContainerBlobStorageTest::shouldWorkWithHiddenFiles()` fails with following error:
> RuntimeException: Failed to retrieve keys: container "" not found.

Have you any idea why this happen?

**Comment by dkarlovi** [comment]
I think we should add a test which fetches keys from a completely empty storage (no containers, no blobs), that might be the issue here. Would need to see the actual exception getting caught there though.
